src/main/java/edu/cornell/kfs/module/purap/rest/resource/CuEinvoiceApiResource.java

```diff
@@ -170,22 +170,47 @@ private VendorAddress getVendorAddress(List<VendorAddress> vendorAddresses) {
+            if (obj instanceof PurchaseOrderItem) {
+                PurchaseOrderItem poItem = (PurchaseOrderItem)obj;
                 Properties lineProps = new Properties();
-                safelyAddProperty(lineProps, CUPurapConstants.AMOUNT.toLowerCase(), purApItem.getTotalAmount().toString());
-                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.QUANTITY, purApItem.getItemQuantity().toString());
-                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.UNIT_OF_MEASURE, purApItem.getItemUnitOfMeasureCode());
-                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.DESCRIPTION, purApItem.getItemDescription());
-                lineProps.put(CUPurapConstants.Einvoice.UNIT_PRICE, purApItem.getItemUnitPrice());
-                lineProps.put(CUPurapConstants.Einvoice.LINE_NUMBER, purApItem.getItemLineNumber());
-                lineProps.put(CUPurapConstants.Einvoice.PART_NUMBER, purApItem.getItemAuxiliaryPartIdentifier());
+                safelyAddProperty(lineProps, CUPurapConstants.AMOUNT.toLowerCase(), poItem.getTotalAmount());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.INVOICED_AMOUNT.toLowerCase(), poItem.getItemInvoicedTotalAmount());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.PO_QUANTITY, poItem.getItemQuantity());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.INVOICE_QUANTITY, poItem.getItemInvoicedTotalQuantity());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.UNIT_OF_MEASURE, poItem.getItemUnitOfMeasureCode());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.DESCRIPTION, poItem.getItemDescription());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.UNIT_PRICE, poItem.getItemUnitPrice());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.LINE_NUMBER, poItem.getItemLineNumber());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.PART_NUMBER, poItem.getItemAuxiliaryPartIdentifier());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.ITEM_TYPE_DESCRIPTION, poItem.getItemType());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.ITEM_TYPE_CODE, poItem.getItemTypeCode());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.PO_ITEM_ID, poItem.getItemIdentifier());
                 poLines.add(lineProps);
             }
         }
         poProperties.put(CUPurapConstants.Einvoice.ITEMS, poLines.toArray());
     }
 
+    private void safelyAddProperty(Properties properties, String key, String value) {
+        properties.put(key, StringUtils.defaultIfBlank(value, KFSConstants.EMPTY_STRING));
+    }
+
+    private void safelyAddProperty(Properties properties, String key, KualiDecimal value) {
+        properties.put(key, ObjectUtils.isNull(value) ? KFSConstants.EMPTY_STRING : value.floatValue());
```

Comment:
Should the KualiDecimal's value just be converted to a String instead of a float, to help avoid the risk of losing data? If you need to convert it to a primitive, I would suggest converting to a "double" instead if you can. The same goes for the equivalent BigDecimal method below.